Neos.Neos/Classes/FrontendRouting/Projection/DocumentUriPathProjection.php

```diff
@@ -549,11 +549,6 @@ private function whenNodePropertiesWereSet(NodePropertiesWereSet $event, EventEn
                 continue;
             }
             $oldUriPath = $node->getUriPath();
-            // homepage -> TODO hacky?
-            if ($oldUriPath === '') {
-                continue;
-            }
```

Comment:
you said

> Instead of checking for empty string, we should check against the site node type

but we dont seem to do this but completely erased the `continue` logic?

Reply:
yep, that was the plan as site nodes were planned to already be caught at !isset($newPropertyValues['uriPathSegment'])
But IIRC properties cannot be undeclared, so we need the check again, right?

Reply:
Added the check

Reply:
> But IIRC properties cannot be undeclared, so we need the check again, right?

yes and that’s why your behat test should also be adjusted the thing ~ is not yet possible

Reply:
See tests (please review ^^) https://github.com/neos/neos-development-collection/pull/4620

Reply:
already did that, is this resolved, ready to merge then?

Reply:
see comment https://github.com/neos/neos-development-collection/pull/4617/files#r1365443531